quic_multiaddr.go

```diff
@@ -28,3 +30,21 @@ func toQuicMultiaddr(na net.Addr) (ma.Multiaddr, error) {
 func fromQuicMultiaddr(addr ma.Multiaddr) (net.Addr, error) {
 	return manet.ToNetAddr(addr.Decapsulate(quicMA))
 }
+
+// this is a bit hacky, but it fixes a problem with "unspecified addresses" (0.0.0.0)
+// appearing in connections. (this breaks libp2p and ipfs assumptions -- listeners
+// may have unspecified addrs, but connections should not -- should use loopback ip).
+func toQuicMultiaddrLocal(na net.Addr) (ma.Multiaddr, error) {
+	qma, err := toQuicMultiaddr(na)
+	if err != nil {
+		return nil, err
+	}
+	rma, err := addrutil.ResolveUnspecifiedAddresses([]ma.Multiaddr{qma}, nil)
+	if err != nil {
+		return nil, err
+	}
+	if len(rma) < 1 {
+		return nil, errors.New("failed to resolve multiaddr")
+	}
+	return rma[0], nil
```

Comment:
this is not the right way to do this, because the first addr here is probably not the right one. We should be able to grab the resolved socket addr from the os.

Reply:
(i believe this is strictly better than what it currently does though)